include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -53,25 +56,58 @@ enum class operational_status : uint8_t
 };
 
 /**
- * Condition which is used to decide if a layout is operational or non-operational.
+ * Parameters for the `is_operational` algorithm.
  */
-enum class operational_condition : uint8_t
+struct is_operational_params
 {
     /**
-     * Even if the I/O pins show kinks, the layout is still considered as operational.
+     * Condition to decide whether a layout is operational or non-operational.
      */
-    TOLERATE_KINKS,
+    enum class operational_condition : uint8_t
+    {
+        /**
+         * Even if the I/O pins show kinks, the layout is still considered as operational.
+         */
+        TOLERATE_KINKS,
+        /**
+         * The I/O pins are not allowed to show kinks. If kinks exist, the layout is considered as non-operational.
+         */
+        REJECT_KINKS
+    };
+
     /**
-     * The I/O pins are not allowed to show kinks. If kinks exist, the layout is considered as non-operational.
+     * Simulation method to determine if the layout is operational or non-operational. There are three possible modes:
+     *
+     * - `SIMULATION_BASED`: This setting does not apply any filtering strategies to determine if the layout is
+     * operational. Instead, it relies solely on physical simulation to make this determination.
+     * - `FILTER_BASED`: This setting does only apply filtering strategies to determine if the layout is
+     * non-operational. If the layout passes all filtering strategies, it is considered operational. This is only an
+     * approximation. It may be possible that the layout is non-operational, but the filtering strategies do not detect
+     * it.
+     * - `FILTER_THEN_SIMULATION`: Before a physical simulation is conducted, the algorithm checks if filtering
+     * strategies have detected whether the layout is non-operational. This only provides any runtime benefits if kinks
+     * are rejected.
      */
-    REJECT_KINKS
-};
-
-/**
- * Parameters for the `is_operational` algorithm.
- */
-struct is_operational_params
-{
+    enum class operational_analysis_strategy : uint8_t
+    {
+        /**
+         * Do not apply filter strategies to determine whether the layout is operational.
+         * Instead, rely solely on physical simulation.
+         */
+        SIMULATION_BASED,
+        /**
+         * Apply filtering exclusively to determine whether the layout is non-operational. If the layout
+         * passes all filter steps, it is considered operational.
+         *
+         * @note This is an extremely fast approximation that may sometimes lead to false positives.
+         */
+        FILTER_BASED,
+        /**
+         * Before a physical simulation is conducted, the algorithm checks if filter strategies can determine that the
+         * layout is non-operational. This only provides any runtime benefits if kinks are rejected.
+         */
+        FILTER_THEN_SIMULATION
```

Comment:
I liked `SIMULATION_ONLY` and `FILTER_ONLY`. I do agree though that `FILTER_THEN_SIMULATION` is nicer.